### PR TITLE
Document removed and obsoleted APIs in ASP.NET Core

### PR DESCRIPTION
**Pull request by scottaddie**
Document the breaking change described at https://github.com/aspnet/Announcements/issues/450

[Internal review page](https://review.docs.microsoft.com/dotnet/core/compatibility/aspnet-core/6.0/obsolete-removed-apis?branch=pr-en-us-22862)

### CODE REVIEW

**Comment by scottaddie** [comment]
@gewarren Ready for review.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready.

File: docs/core/compatibility/6.0.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ If you're migrating an app to .NET 6.0, the breaking changes listed here might a
 ## ASP.NET Core
 
 - [Nullable reference type annotations changed](aspnet-core/6.0/nullable-reference-type-annotations-changed.md)
+- [Obsoleted and removed APIs](aspnet-core/6.0/obsolete-removed-apis.md)
```
Comment (captainsafia) [inline docs/core/compatibility/6.0.md:16]: There's an additional set of modified APIs at https://github.com/aspnet/Announcements/issues/446.

Comment (scottaddie) [reply docs/core/compatibility/6.0.md:16]: Those APIs were already documented in a separate doc: https://docs.microsoft.com/dotnet/core/compatibility/aspnet-core/6.0/razor-engine-apis-obsolete?branch=pr-en-us-22862

**Review by captainsafia — APPROVED** [review]

**Pull request merged**